packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -423,12 +425,15 @@ function validateClassAttr(vnode: VBaseElement, elm: Element, renderer: Renderer
     // are rendered during SSR. This needs to be accounted for when validating.
     if (scopedToken) {
         if (!isUndefined(className)) {
-            className = `${scopedToken} ${className}`;
+            className = `${scopedToken} ${className} ${stylesheetTokenHost}`.trim();
         } else if (!isUndefined(classMap)) {
             classMap = {
                 ...classMap,
                 [scopedToken]: true,
+                ...(stylesheetTokenHost && { [stylesheetTokenHost]: true }),
             };
+        } else {
+            className = `${scopedToken} ${stylesheetTokenHost}`.trim();
```

Comment:
This is to account for the scenario when no class attributes are listed on the element.

Reply:
Same question about `trim` here.